packages/lesswrong/components/community/modules/SearchResultsMap.tsx

```diff
@@ -0,0 +1,121 @@
+    fill: theme.palette.individual,
+    opacity: 0.8,
+    cursor: 'pointer'
+  },
+  popupAddress: {
+    ...theme.typography.commentStyle,
+    color: theme.palette.grey[600],
+    fontSize: 12,
+    fontStyle: 'italic',
+  },
+  popupBio: {
+    ...theme.typography.commentStyle,
+    color: theme.palette.grey[800],
+    fontSize: 14,
+    lineHeight: '1.8em',
+    whiteSpace: 'pre-wrap',
+    display: '-webkit-box',
+    "-webkit-line-clamp": 3,
+    "-webkit-box-orient": 'vertical',
+    overflow: 'hidden',
+    marginTop: 12,
+  }
+}))
+
+const defaultCenter = {lat: 39.5, lng: -43.636047}
+
+const SearchResultsMap = ({center = defaultCenter, zoom = 2, hits, classes}: {
+  center: {lat: number, lng: number},
+  zoom: number,
+  hits: Array<Hit<AlgoliaUser>>,
+  classes: ClassesType,
+}) => {
+  const [activeResultId, setActiveResultId] = useState('')
+  
+  const [viewport, setViewport] = useState({
+    latitude: center.lat,
+    longitude: center.lng,
+    zoom: zoom
+  })
+  
+  // when getting the location from the browser, we want to re-center the map
+  useEffect(() => {
+    setViewport({
+      latitude: center.lat,
+      longitude: center.lng,
+      zoom: zoom
+    })
+  }, [center.lat, center.lng, zoom])
+  
+  const { StyledMapPopup } = Components
+  
+  const isEAForum = forumTypeSetting.get() === 'EAForum'
+  
+  return <div className={classes.root}>
+      <Helmet>
+        <link href='https://api.tiles.mapbox.com/mapbox-gl-js/v1.3.1/mapbox-gl.css' rel='stylesheet' />
+      </Helmet>
+      <ReactMapGL
+        {...viewport}
+        width="100%"
+        height="100%"
+        mapStyle={isEAForum ? undefined : "mapbox://styles/habryka/cilory317001r9mkmkcnvp2ra"}
+        onViewportChange={viewport => setViewport(viewport)}
+        mapboxApiAccessToken={mapboxAPIKeySetting.get() || undefined}
+      >
+        {hits.map(hit => {
+          if (!hit._geoloc) return null
+          
+          return <React.Fragment key={hit._id}>
+            <Marker
+              latitude={hit._geoloc.lat}
+              longitude={hit._geoloc.lng}
+              offsetLeft={-8}
+              offsetTop={-20}
+            >
+              <PersonIcon className={classes.icon} onClick={() => setActiveResultId(hit._id)} />
+            </Marker>
+            {(activeResultId === hit._id) && <StyledMapPopup
+              lat={hit._geoloc.lat}
+              lng={hit._geoloc.lng}
+              link={userGetProfileUrl(hit)}
+              title={hit.displayName}
+              onClose={() => setActiveResultId('')}
+              hideBottomLinks
+            >
+              <div className={classes.popupAddress}>{hit.mapLocationAddress}</div>
+              <div className={classes.popupBio}>{hit.bio}</div>
+            </StyledMapPopup>}
+          </React.Fragment>
+        })}
+      </ReactMapGL>
+  </div>
+}
+
+const SearchResultsMapComponent = registerComponent("SearchResultsMap", connectHits(SearchResultsMap), { styles });
```

Comment:
```suggestion
// connectHits is probably nothing but a consumer acting as a HoC, like this:
// const connectHits = (C) => { const hits = useHits(); return (args) => C({...args, hits}); }
// It consumes the data provided by InstantSearch, which should be a parent of us
const SearchResultsMapComponent = registerComponent("SearchResultsMap", connectHits(SearchResultsMap), { styles });
```

Reply:
I think I believe you that they don't expose an api for using the hypothesized `useHits` directly. Might still be worth asking their support about, because it would be more idiomatic.